<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>899</width>
    <height>550</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777212</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>3dbpp</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLineEdit" name="in_box_w">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>500</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="text">
     <string>150</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_binpacking">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>520</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>Binpacking</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_load">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>Load</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_showorder">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>520</y>
      <width>71</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>Show Order</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_next_order">
    <property name="geometry">
     <rect>
      <x>365</x>
      <y>500</y>
      <width>16</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>&gt;</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="in_box_h">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>500</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="text">
     <string>110</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>500</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>Box size(w*h*d)</string>
    </property>
   </widget>
   <widget class="GLWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>40</y>
      <width>500</width>
      <height>500</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_remove">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>30</y>
      <width>55</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>Remove</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="in_box_d">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>500</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="text">
     <string>50</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_previous_order">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>500</y>
      <width>16</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>&lt;</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_add_box">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>500</y>
      <width>31</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>Add</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_save">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>30</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>Save</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_clearall">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>30</y>
      <width>55</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>Clear All</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="in_bin_d">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>520</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="text">
     <string>225</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="in_bin_h">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>520</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="text">
     <string>220</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>520</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>Bin size(w*h*d)</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="in_bin_w">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>520</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="text">
     <string>330</string>
    </property>
   </widget>
   <widget class="QTreeWidget" name="treeWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>371</width>
      <height>431</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>0</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777215</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="sizeAdjustPolicy">
     <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
    </property>
    <property name="autoScrollMargin">
     <number>14</number>
    </property>
    <property name="dragEnabled">
     <bool>true</bool>
    </property>
    <property name="dragDropOverwriteMode">
     <bool>true</bool>
    </property>
    <property name="dragDropMode">
     <enum>QAbstractItemView::InternalMove</enum>
    </property>
    <property name="defaultDropAction">
     <enum>Qt::TargetMoveAction</enum>
    </property>
    <property name="selectionBehavior">
     <enum>QAbstractItemView::SelectRows</enum>
    </property>
    <property name="sortingEnabled">
     <bool>false</bool>
    </property>
    <attribute name="headerVisible">
     <bool>true</bool>
    </attribute>
    <attribute name="headerDefaultSectionSize">
     <number>40</number>
    </attribute>
    <attribute name="headerHighlightSections">
     <bool>false</bool>
    </attribute>
    <attribute name="headerMinimumSectionSize">
     <number>40</number>
    </attribute>
    <attribute name="headerShowSortIndicator" stdset="0">
     <bool>false</bool>
    </attribute>
    <attribute name="headerStretchLastSection">
     <bool>true</bool>
    </attribute>
    <column>
     <property name="text">
      <string>No.</string>
     </property>
     <property name="font">
      <font>
       <family>Calibri</family>
      </font>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>w</string>
     </property>
     <property name="toolTip">
      <string/>
     </property>
     <property name="font">
      <font>
       <family>Calibri</family>
      </font>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>h</string>
     </property>
     <property name="font">
      <font>
       <family>Calibri</family>
      </font>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>d</string>
     </property>
     <property name="font">
      <font>
       <family>Calibri</family>
      </font>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Name</string>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>r</string>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>g</string>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>b</string>
     </property>
     <property name="textAlignment">
      <set>AlignCenter</set>
     </property>
    </column>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>500</y>
      <width>25</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>mm</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>520</y>
      <width>25</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>mm</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_reset">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>10</y>
      <width>50</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_packMethod">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>10</y>
      <width>75</width>
      <height>22</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <item>
     <property name="text">
      <string>General</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Robot</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>10</y>
      <width>55</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>nodelimit</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="in_node_limit">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>10</y>
      <width>40</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="text">
     <string>0</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>10</y>
      <width>55</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>iterlimit</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="in_iter_limit">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>10</y>
      <width>40</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="text">
     <string>0</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>10</y>
      <width>55</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>timelimit</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="in_time_limit">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>10</y>
      <width>40</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="text">
     <string>1</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>10</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>sec</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_filename">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>441</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>filename</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_order">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>500</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>0</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_moveup">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>30</y>
      <width>71</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>move up</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_movedown">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>30</y>
      <width>71</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Calibri</family>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="text">
     <string>move down</string>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
